### DISCUSSION TITLE
How to Read the remote host IP Address from a request

### DISCUSSION THREAD

**Discussion by lobstromo** [emmett-framework/emmett#420] [Q&A] [answered]
I'm trying to read the IP Address of the remote host from a request but it always seems to just return 127.0.0.1.
For a sanity check I created a hello world app with bottle.
Here's my code:

from emmett import App, request

app = App(__name__)

@app.route('/')
async def index():
	return request.client

What am I doing wrong?

**Comment by gi0baro** [comment]
@lobstromo the `Request.client` property checks 2 different headers fo return the address, specifically:
- `X-Forwarded-For` 
- `Remote-Addr`

In case none of those contains valid results, the property fall-backs to `127.0.0.1`.
So it really depends where your application is running. In case your stack doesn't have values for those headers and you have the information in a specific one, you can still call `Request.headers['your-header']`.

**Comment by lobstromo** [comment]
I was able to get this working by updating request.py.
It might not be the proper way to do it but I am able to see the remote host's ip address now.

 75 class Request(ScopeWrapper):                                                                        
 76     __slots__ = ['_now', 'method']                                                                  
 77                                                                                                     
 78     def __init__(                                                                                   
 79         self,                                                                                       
 80         scope: Scope,                                                                               
 81         receive: Receive,                                                                           
 82         send: Send,                                                                                 
 83         max_content_length: Optional[int] = None,                                                   
 84         body_timeout: Optional[int] = None                                                          
 85     ):                                                                                              
 86         super().__init__(scope, receive, send)                                                      
 87         self.max_content_length = max_content_length                                                
 88         self.body_timeout = body_timeout                                                            
 89         self._now = datetime.utcnow()                                                               
 90         self.method = scope['method']                                                                                                                
 **91         self.client = scope['client'][0]**

**Comment by gi0baro** [comment, marked as answer]
@lobstromo 2.4.4 should now implement this properly

**Reply by lobstromo** [reply]
Groovy! Thanks!!
